MDANSE_GUI/Src/MDANSE_GUI/MolecularViewer/MolecularViewer.py

```diff
@@ -274,270 +252,485 @@ def add_text(text, coord):
-        actor.GetProperty().SetColor(colour)
-
-        actor.SetPosition(position)
-        actor.SetScale(scale)
-        # make center sphere metallic and semi-transparent
-        if filename == "center_sphere.stl":
-            actor.GetProperty().SetOpacity(
-                0.5
-            )  # Make the center sphere semi-transparent
-            actor.GetProperty().SetSpecular(1)
-            actor.GetProperty().SetSpecularPower(100)
-            actor.GetProperty().SetMetallic(1.0)
-            actor.GetProperty().SetRoughness(0.2)
-
-        individual_transform = vtk.vtkTransform()
-        individual_transform.Identity()
-        individual_transform.Translate(0.0, 0.5, 0.5)
-        individual_transform.Scale(0.5, 0.5, 0.5)
-        actor.SetUserTransform(individual_transform)
-
-        return actor
-
-    def load_trajectory_placeholder_3d_model(self):
-        """Loads a 3D blender model of mdanse into the viewer."""
-
-        self._actors = vtk.vtkAssembly()
-        spheres = [
-            ("center_sphere.stl", (0.6, 0.6, 0.6), 0.4, (0, 0, 0)),  # grey center
-            ("yellow_sphere.stl", (0.2, 0.85, 0.25), 0.4, (0, 0, 0)),
-            ("blue_sphere.stl", (0.2, 0.35, 0.85), 0.4, (0, 0, 0)),
-            ("red_sphere.stl", (0.85, 0.24, 0.2), 0.4, (0, 0, 0)),
-        ]
-
-        for filename, colour, position, scale in spheres:
-            actor = self.create_actor_for_placeholder_3d_model(
-                filename, colour, position, scale
+        if self.atom_label_type == "index":
+            labels = list(range(self._reader._n_atoms))
+        elif self.atom_label_type == "label":
+            label_dict = self._reader._trajectory.chemical_system._labels
+            if not label_dict:
+                return
+            keys = more_itertools.run_length.decode(
+                ((k, len(v)) for k, v in label_dict.items())
+            )
+            labels = sorted(keys, key=label_dict.__getitem__)
+        elif self.atom_label_type == "atom":
+            labels = self._atoms
+        elif self.atom_label_type == "molecule":
+            label_dict = self._reader._trajectory.chemical_system._clusters
+            if not label_dict:
+                return
+            label_dict = {
+                k: list(more_itertools.collapse(v)) for k, v in label_dict.items()
+            }
+            keys = more_itertools.run_length.decode(
+                ((k, len(v)) for k, v in label_dict.items())
             )
-            self._actors.AddPart(actor)
+            labels = sorted(keys, key=label_dict.__getitem__)
+        else:
+            return
 
-        self._renderer.AddActor(self._actors)
+        if self._current_coords is None or len(labels) != self._reader._n_atoms:
+            return
 
-        # side view of 3d model
-        self._camera.SetPosition(20, 0, 0)
-        self._camera.SetFocalPoint(0, 0, 0)
-        self._camera.SetViewUp(0, 0, 1)
+        atom_label_actors = []
```

Comment:
Why bother with the temp?

```suggestion
        self.atom_label_actors = []
```